subworkflows/nf-core/fasta_gxf_busco_plot/main.nf

```diff
@@ -0,0 +1,172 @@
+    ch_busco_assembly_inputs                    = ch_fasta
+                                                | combine(
+                                                    Channel.of(val_mode)
+                                                )
+                                                | combine(
+                                                    Channel.fromList(val_lineages)
+                                                )
+                                                | map { meta, fasta, mode, lineage ->
+                                                    [
+                                                        meta + [ mode:mode, lineage:lineage ],
+                                                        fasta, mode, lineage
+                                                    ]
+                                                }
+                                                | combine(
+                                                    ch_db_path
+                                                )
+                                                | combine(
+                                                    ch_config_path
+                                                )
+
+    BUSCO_ASSEMBLY(
+        ch_busco_assembly_inputs.map { meta, fasta, mode, lineage, db, config -> [ meta, fasta ] },
+        ch_busco_assembly_inputs.map { meta, fasta, mode, lineage, db, config -> mode },
+        ch_busco_assembly_inputs.map { meta, fasta, mode, lineage, db, config -> lineage },
+        ch_busco_assembly_inputs.map { meta, fasta, mode, lineage, db, config -> db ?: [] },
+        ch_busco_assembly_inputs.map { meta, fasta, mode, lineage, db, config -> config ?: [] }
+    )
+
+    ch_assembly_batch_summary                   = BUSCO_ASSEMBLY.out.batch_summary
+    ch_assembly_short_summaries_txt             = BUSCO_ASSEMBLY.out.short_summaries_txt
+    ch_assembly_short_summaries_json            = BUSCO_ASSEMBLY.out.short_summaries_json
+    ch_assembly_full_table                      = BUSCO_ASSEMBLY.out.full_table
+    ch_versions                                 = ch_versions.mix(BUSCO_ASSEMBLY.out.versions.first())
+
+    // MODULE: BUSCO_GENERATEPLOT as PLOT_ASSEMBLY
+    ch_assembly_plot_summary                    = ch_assembly_short_summaries_txt
+                                                | map { meta, txt ->
+                                                    def lineage_name = meta.lineage.split('_odb')[0]
+                                                    [
+                                                        "short_summary.specific.${meta.lineage}.${meta.id}_${lineage_name}.txt",
+                                                        txt.text
+                                                    ]
+                                                }
+                                                | collectFile
+
+    PLOT_ASSEMBLY( ch_assembly_plot_summary.collect() )
```

Comment:
You're already using `collectFile` so you don't need `collect` I think.

Reply:
`collectFile` collects the file. `collect` collects all the sample x lineage files so that they can be plotted on a single figure.